src/server/req_delete.c

```diff
@@ -201,103 +201,169 @@ acct_del_write(char *jid, job *pjob, struct batch_request *preq, int nomail)
  *		done by req_deletejob()
  *
  * @param[in]	preq	- Batch request structure.
+ * @param[in]	pjob	- pointer to job
+ * @param[in]	parent	- pointer to parent job for sub job
+ * @param[in]	jt	- job type
  *
  * @return	int
  * @retval	TRUE	- Job history  has been purged
  * @retval	FALSE	- Job is not a history job
  */
 int
-check_deletehistoryjob(struct batch_request * preq)
+check_deletehistoryjob(struct batch_request * preq, job *pjob, job *parent, int jt)
 {
-	job *histpjob;
-	job *pjob;
-	int historyjob;
-	int histerr;
 	char *jid;
+	job *psjob;
+	char *sjid;
+	char *pc;
+	int i,j;
+	int x, y, z;
+
 	jid = preq->rq_ind.rq_delete.rq_objname;
 
-	histpjob = find_job(jid);
-	/*
-	 * if histpjob returns NULL then we can be sure its a array subjob or a range of subjobs . If the array
-	 * subjob or range of subjobs are in a history state then reject the request as we cant delete history
-	 * of array subjobs
-	 */
-	if (histpjob == NULL) {
-		pjob = find_arrayparent(jid);
-		if ((histerr = svr_chk_histjob(pjob))) {
-			req_reject(PBSE_NOHISTARRAYSUBJOB, 0, preq);
-			return TRUE;
-		} else {
-			/*
-			 * Job is in a Non Finished state . It must be terminated and then its history
-			 *  should be purged .
-			 */
-			return FALSE;
-		}
+	if (PBSE_HISTJOBID != svr_chk_histjob(pjob))
+		return FALSE;
+
+
+	/* Issue history job delete request to remote server if job is moved. */
+	if (pjob->ji_qs.ji_state == JOB_STATE_MOVED)
```

Comment:
Are we deleting the history of subjobs with this change ? It seems this was not possible earlier. If yes then EDD should reflect this.

Reply:
This change actually enabled deleting history of subjob, However the scope of this PR is limited to subjob surviving restart, hence with rc 5 commit, I have removed the capability